test failed
```
//pkg/volume/photon_pd:go_default_test 0.00s
bazel test //pkg/volume/photon_pd:go_default_test
E0808 15:21:59.540258       9 attacher.go:254] Error checking if persistent disk ("000-000-000") is already attached to current node ("instance"). Will continue and try detach anyway. err=Fake DiskIsAttached error
E0808 15:21:59.540351       9 attacher.go:254] Error checking if persistent disk ("000-000-000") is already attached to current node ("instance"). Will continue and try detach anyway. err=Fake DiskIsAttached error
E0808 15:21:59.540365       9 attacher.go:264] Error detaching volume "000-000-000": Fake detach error
--- FAIL: TestAttachDetach (0.00s)
	attacher_test.go:290: Unexpected DiskIsAttached call!
	attacher_test.go:160: Test "Attach_Positive" succeeded
	attacher_test.go:290: Unexpected DiskIsAttached call!
	attacher_test.go:160: Test "Attach_Positive_AlreadyAttached" succeeded
	attacher_test.go:160: Test "Detach_Positive" succeeded
	attacher_test.go:160: Test "Detach_Positive_AlreadyDetached" succeeded
	attacher_test.go:160: Test "Detach_Positive_CheckFails" succeeded
	attacher_test.go:160: Test "Detach_Negative" succeeded
FAIL
Use --strategy=TestRunner=standalone to disable sandboxing for the failing actions.
```

--- a/pkg/volume/photon_pd/attacher.go
+++ b/pkg/volume/photon_pd/attacher.go
@@ -67,14 +67,21 @@ func (attacher *photonPersistentDiskAttacher) Attach(spec *volume.Spec, nodeName
 		glog.Errorf("Photon Controller attacher: Attach failed to get volume source")
 		return "", err
 	}
+	attached, err := attacher.photonDisks.DiskIsAttached(volumeSource.PdID, nodeName)
 
-	glog.V(4).Infof("Photon Controller: Attach disk called for host %s", hostName)
-
-	// TODO: if disk is already attached?
-	err = attacher.photonDisks.AttachDisk(volumeSource.PdID, nodeName)
 	if err != nil {
-		glog.Errorf("Error attaching volume %q to node %q: %+v", volumeSource.PdID, nodeName, err)
-		return "", err
+		glog.Warningf("Photon Controller: couldn't check if disk is Attached for host %s, will try attach disk: %+v", hostName, err)
+		attached = false
+	}
+
+	if !attached {
+		glog.V(4).Infof("Photon Controller: Attach disk called for host %s", hostName)
+
+		err = attacher.photonDisks.AttachDisk(volumeSource.PdID, nodeName)
+		if err != nil {
+			glog.Errorf("Error attaching volume %q to node %q: %+v", volumeSource.PdID, nodeName, err)
+			return "", err
+		}
 	}
 
 	PdidWithNoHypens := strings.Replace(volumeSource.PdID, "-", "", -1)

--- a/pkg/volume/photon_pd/attacher_test.go
+++ b/pkg/volume/photon_pd/attacher_test.go
@@ -78,14 +78,14 @@ func TestAttachDetach(t *testing.T) {
 	nodeName := types.NodeName("instance")
 	readOnly := false
 	spec := createVolSpec(diskName, readOnly)
-	attachError := errors.New("Fake attach error")
 	detachError := errors.New("Fake detach error")
 	diskCheckError := errors.New("Fake DiskIsAttached error")
 	tests := []testcase{
 		// Successful Attach call
 		{
-			name:   "Attach_Positive",
-			attach: attachCall{diskName, nodeName, nil},
+			name:           "Attach_Positive",
+			diskIsAttached: diskIsAttachedCall{diskName, nodeName, false, diskCheckError},
+			attach:         attachCall{diskName, nodeName, nil},
 			test: func(testcase *testcase) (string, error) {
 				attacher := newAttacher(testcase)
 				return attacher.Attach(spec, nodeName)
@@ -93,15 +93,16 @@ func TestAttachDetach(t *testing.T) {
 			expectedDevice: "/dev/disk/by-id/wwn-0x000000000",
 		},
 
-		// Attach call fails
+		// Disk is already attached
 		{
-			name:   "Attach_Negative",
-			attach: attachCall{diskName, nodeName, attachError},
+			name:           "Attach_Positive_AlreadyAttached",
+			diskIsAttached: diskIsAttachedCall{diskName, nodeName, false, diskCheckError},
+			attach:         attachCall{diskName, nodeName, nil},
 			test: func(testcase *testcase) (string, error) {
 				attacher := newAttacher(testcase)
 				return attacher.Attach(spec, nodeName)
 			},
-			expectedError: attachError,
+			expectedDevice: "/dev/disk/by-id/wwn-0x000000000",
 		},
 
 		// Detach succeeds

